### DISCUSSION TITLE
Can't play music

### DISCUSSION THREAD

**Discussion by chen-junl** [KRTirtho/spotube#1163] [Q&A]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Current Behavior

When I search for Jay's songs and click play, the icon keeps loading and the music cannot be played.
1. 

### Expected Behavior

 play music

### Steps to reproduce

When I search for Jay's songs and click play, the icon keeps loading and the music cannot be played.

![image](https://github.com/KRTirtho/spotube/assets/38237837/62db235a-fa90-4092-a025-50350671b92e)

![image](https://github.com/KRTirtho/spotube/assets/38237837/b31ba33c-5e66-463a-8969-bcaf88d55f8b)

1. 

### Operating System

Windows 11 22631.3085

### Spotube version

v3.4.0

### Installation source

GitHub Releases (Binary)

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by chen-junl** [comment]
============================== CATCHER 2 LOG ==============================
------- DEVICE INFO -------
computerName: DESKTOP-JLRLRGU
numberOfCores: 8
systemMemoryInMegabytes: 16384
------- APP INFO -------
environment: release
version: 3.4.0
appName: spotube
buildNumber: 
packageName: spotube
---------- ERROR ----------
FormatException: Unexpected character (at character 1)
Too many requests
^
------- STACK TRACE -------
#0      _ChunkedJsonParser.fail (dart:convert-patch/convert_patch.dart:1383)
#1      _ChunkedJsonParser.parseNumber (dart:convert-patch/convert_patch.dart:1250)
#2      _ChunkedJsonParser.parse (dart:convert-patch/convert_patch.dart:915)
#3      _parseJson (dart:convert-patch/convert_patch.dart:35)
#4      JsonDecoder.convert (dart:convert/json.dart:610)
#5      JsonCodec.decode (dart:convert/json.dart:216)
#6      SpotifyApiBase.handleErrors (package:spotify/src/spotify_base.dart:232)
#7      SpotifyApiBase._requestWrapper (package:spotify/src/spotify_base.dart:212)
<asynchronous suspension>
#8      SpotifyApiBase._getImpl (package:spotify/src/spotify_base.dart:178)
<asynchronous suspension>
#9      Pages.getPage (package:spotify/src/endpoints/endpoint_paging.dart:236)
<asynchronous suspension>
#10     PlaylistQueries.featured.<anonymous closure> (package:spotube/services/queries/playlist.dart:255)
<asynchronous suspension>
#11     _AsyncCompleter.complete (dart:async/future_impl.dart:41)
<asynchronous suspension>
#12     Retryer.retryOperation (package:fl_query/src/core/mixins/retryer.dart:17)
<asynchronous suspension>
#13     CancelableCompleter.complete.<anonymous closure> (package:async/src/cancelable_operation.dart:425)
<asynchronous suspension>
------- CUSTOM INFO -------

**Comment by KRTirtho** [comment]
It seems to be working fine.

<img width="1206" alt="image" src="https://github.com/KRTirtho/spotube/assets/61944859/e911ca0e-1bc1-49a9-8a5c-310924f7b50a">

**Comment by nepalivai** [comment]
That also seems to happen on my device I can't find the exact reason when that occurs
`(It occurs when i switch music rapidly like one after another/ playing a playlist for a certain {little long time} then switching song in or out of the play list) ` **Sometimes it happens sometime it does not**
but I have a quick fix just exit/ force close the app and reopen it. That should fix it temporarily.

**Comment by raycw** [comment]
I have similar issue which should be the spotify returns "too many requests". I think it may be the api returns http 429 and the client can't handle it. But I'm not familiar with spotify api so I have no idea about it.

**Reply by KRTirtho** [reply]
Spotify API isn't related with this because as soon as we get track from Spotify, it tries to fetch the Audio source when you click play.

**Reply by raycw** [reply]
I have checked the log, it returns "Too many requests" from package [spotify](https://github.com/rinukkusu/spotify-dart/blob/ca66bd116a496e0b71c6160feaca417463bab3a9/lib/src/spotify_base.dart#L234). You may refer to the screenshot.
![image](https://github.com/KRTirtho/spotube/assets/737986/688ffa8f-b135-44fb-8b17-212ca340c9f8)
My problem is the home screen can't load the "Genres". 
![image](https://github.com/KRTirtho/spotube/assets/737986/0a7b0c59-231a-489f-9a3e-a725cb890f22)
And I check [spotify api doc](https://developer.spotify.com/documentation/web-api/concepts/rate-limits), I think the app has reached spotify api rate limited.

**Comment by chen-junl** [comment]
I'm not sure if it's my network problem, are our settings the same?

![image](https://github.com/KRTirtho/spotube/assets/38237837/978ae746-58a0-4e96-a7a5-f1d895a65567)

![image](https://github.com/KRTirtho/spotube/assets/38237837/4d923571-d14c-4a94-b6f0-b7e7114cac1c)
